### PR TITLE
PP-842: fix underpriced transactions

### PR DESCRIPTION
**Pull request by antomor**
## What

- fix broken tests
- include tests to cover server fixes

## Why

- the tests were broken due to some changes in the server
- the tests don't cover a bug we recently found

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
     "@nomiclabs/hardhat-ethers": "^2.2.1",
     "@rsksmart/rif-relay-client": "github:rsksmart/rif-relay-client",
     "@rsksmart/rif-relay-contracts": "github:rsksmart/rif-relay-contracts",
-    "@rsksmart/rif-relay-server": "github:rsksmart/rif-relay-server",
+    "@rsksmart/rif-relay-server": "github:rsksmart/rif-relay-server#PP-842/fix-underpriced-transactions",
```
Comment (antomor) [inline package.json:93]: This needs to be updated when the [related PR](https://github.com/rsksmart/rif-relay-server/pull/128) is merged.

File: package.json [hunk 2]
```diff
@@ -90,7 +90,7 @@
     "@nomiclabs/hardhat-ethers": "^2.2.1",
     "@rsksmart/rif-relay-client": "github:rsksmart/rif-relay-client",
     "@rsksmart/rif-relay-contracts": "github:rsksmart/rif-relay-contracts",
-    "@rsksmart/rif-relay-server": "github:rsksmart/rif-relay-server#PP-842/fix-underpriced-transactions",
+    "@rsksmart/rif-relay-server": "github:rsksmart/rif-relay-server#fix/resendTransaction",
```
Comment (antomor) [inline package.json:93]: To be reverted once the related PR is merged

Comment (franciscotobar) [reply package.json]: reverted

**Review by franciscotobar — APPROVED** [review]
LGTM

**Pull request merged**